docs/about/coding_guidelines.md

```diff
@@ -0,0 +1,68 @@
+---
+id: about_coding_guidelines
+title:  "ZIO Coding Guidelines"
+---
+
+These are coding guidelines strictly for ZIO contributors for ZIO projects and 
+not general conventions to be applied by the Scala community.
+
+Additionally, bear in mind that, although we try to enforce these rules to the 
+best of our ability, both via automated rules (scalafix) and strict reviewing 
+processes, it is both possible to find existing code that does not comply to 
+these rules. If that is the case, we would be extremely grateful if you could 
+make a contribution, by proving a fix to said issue.
+
+Last, but not least, these rules are continuously evolving and as such, 
+refer to them once in a while when in doubt.
+
+
+### Defining classes and traits
+
+1. Value classes must be final and extend AnyVal.
+2. Method extension classes must be final and extend `AnyVal`.
+3. Sealed traits that are ADTs should extend `Product` and `Serializable`.
+4. Regular traits and sealed trait that do not form ADTS should extend `Serializable` but not `Product`.
+5. Multiple parameter classes that are ADT should be case class and not extend AnyVal
```

Comment:
I'm not sure what this rule means, but you can't extend `AnyVal` on a class with more than one constructor parameter anyway, so that part is redundant.

Reply:
True, the compiler will not allow you to do that.

Reply:
Yes, I think if something is just impossible to do we don't need it here. This is for things that people could do that we don't want them to do.

Reply:
Right, I'd delete 5 for this reason. 👍 

The other ones look great!